src/torcontrol.cpp

```diff
@@ -449,6 +449,16 @@ void TorController::auth_cb(TorControlConnection& conn, const TorControlReply& r
 {
     if (reply.code == 250) {
         LogPrint("tor", "tor: Authentication succesful\n");
+
+        // Now that we know Tor is running setup the proxy for onion addresses
+        // if -onion isn't set to something else.
+        if (GetArg("-onion", "") == "") {
+            bool proxyRandomize = GetBoolArg("-proxyrandomize", true);
```

Comment:
Nit: Probably, proxyRandomize should be always true if we know this is Tor. The option exists because people may want to disable it for non-tor SOCKS5 proxies, there's not really a reason to disable it for Tor.